src/main/java/org/hsharan/components/Ladder.java

```diff
@@ -0,0 +1,40 @@
+package org.hsharan.components;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Ladder {
+    private int start;
+    private int end;
+
+    private static Map<Integer,Ladder> ladders=new HashMap<>();
+
+    public int getStart() {
+        return start;
+    }
+
+    public void setStart(int start) {
+        this.start = start;
+    }
+
+    public int getEnd() {
+        return end;
+    }
+
+    public void setEnd(int end) {
+        this.end = end;
+    }
+
+    public Ladder(int start, int end) throws Exception {
+        if(ladders.containsKey(start) && Snake.getSnake(start)!=null){
```

Comment:
if there is an existing ladder at start index OR there is a Snake at starting index, the this is not possible. Should it not be Or in place of And ?